### ISSUE TITLE
Calendar - days of the week text doesn't meet required contrast ratio

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#7423]
### Expected Behavior
Calendar days of the week text should meet accessibility contrast ratios.

### Actual Behavior
There is a 50% opacity applied to the days of the week above the calendar. This text color with the opacity doesn't meet contrast requirements.

### URL, screen shot, or Codepen exhibiting the issue
<img width="782" alt="Screenshot 2024-11-08 at 2 43 13 PM" src="https://github.com/user-attachments/assets/b58b7e6e-fa5f-412b-8f32-cad677185273">

### Steps to Reproduce

https://storybook.grommet.io/?path=/story/visualizations-calendar-fill--fill

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by NOMADE55** [comment]
Seems to only be the days outside of the month. 
Which makes sense because the opacity _has_ to be low if these are from previous/next month.

**Comment by jcfilben** [comment]
Agree, it makes sense visually to have the opacity low to communicate that the days are outside of the month but then we run into issues because these controls are still interactive and therefore need to meet contrast requirements for WCAG. Not sure what the correct answer is here but what we have at the moment isn't meeting accessibility requirements.

**Comment by NOMADE55** [comment]
Maybe a sr-only span with a comment remarking how these dates are outside of the month.
Other than that, I'm not sure if this is something that can be solved without seriously impacting the design.

**Comment by luketa8** [comment]
I think we have two options here:

### Option #1

Change -> `opacity: 0,5;` to `opacity: 0,81;`.

This passes WCAG AA and AAA Text Contrast with 4.6:1.

### Option #2

Use `color.text.weak` Token: `#757575`

This passes WCAG AA and AAA Text Contrast with 4.6:1.

**Comment by luketa8** [comment]
Tagging @jcfilben @oliverHPE and @taysea for review. Thanks!

**Comment by jcfilben** [comment]
Thanks for sharing the options, I feel like it is a little hard to distinguish the color of the dates inside the month vs the dates outside of the month. I know that that we don't have a lot of options here to make sure we meet color contrast while also making the dates outside the month visually distinct from the rest of the days. I'm wondering how other component libraries are handling this issue? Is there anything we could learn from their approaches

It might also be helpful to show examples of these colors with a calendar where the dates are closer together (something like what we see here: https://storybook.grommet.io/?path=/story/visualizations-calendar-multiple--multiple)

**Comment by luketa8** [comment]
Thanks for the feedback, Jessica. 

After a bit of research, I think we have two solid options that meet the updated [Success Criteria](https://www.figma.com/design/OBxHRrYylrg1IyNetpK7OF/%237423---Calendar-contrast?node-id=28-101&t=G6Y0Zp5ql1zjUy7f-4):

### [Option #1](https://www.figma.com/design/OBxHRrYylrg1IyNetpK7OF/%237423---Calendar-contrast?node-id=28-80&t=G6Y0Zp5ql1zjUy7f-4):
Use `color.text.strong` for days of current month, and `color.text.weak` for previous/following month. 

### [Option #2](https://www.figma.com/design/OBxHRrYylrg1IyNetpK7OF/%237423---Calendar-contrast?node-id=28-80&t=G6Y0Zp5ql1zjUy7f-4):
Show only days from selected month.


I think the best solution is Option #2.

**Comment by jcfilben** [comment]
Thanks for the updated options, added some comments in [the Figma file](https://www.figma.com/design/OBxHRrYylrg1IyNetpK7OF/%237423---Calendar-contrast?node-id=28-80&t=RyZGBoYJSR1uEMu4-0)

**Comment by luketa8** [comment]
Thanks for the feedback, Jessica. 

Here's an updated set of [current colors](https://www.figma.com/design/OBxHRrYylrg1IyNetpK7OF/%237423---Calendar-contrast?node-id=37-6&t=Tkzt92fTvQnqm5Vu-4) for both modes and [viable options for each case.](https://www.figma.com/design/OBxHRrYylrg1IyNetpK7OF/%237423---Calendar-contrast?node-id=45-31&t=Tkzt92fTvQnqm5Vu-4) I've also added my choice for each case, not sure if applicable in code.

**Comment by jcfilben** [comment]
Cool I like the following:
Grommet theme
light mode - option 2
dark mode - stay with current

HPE theme
light mode - option 1
dark mode - option 1
I like option 1 over option 2 because I think it is nice to still allow for showing the dates that are outside of the month and leave it up to the designer to decide what works best for their use case (show dates vs not show dates outside the month). It is also a bit cleaner from the code side to go with option 1

**Comment by luketa8** [comment]
Perfect, thanks Jessica. Closing this task and opening a follow up dev ticket.


# Updates on the calendar text colors:

## Grommet theme

### Light mode:

 - Leading/trailing month: `text-xweak` (#666666)
 - Current months: `text-strong` (#000000)
 
### Dark mode

No changes

## HPE theme

### Light mode:

 - Leading/trailing month: `color.text.weak` (#757575)
 - Current months: `color.text.strong` (#333333)
 
### Dark mode

 - Leading/trailing month: `color.text.weak` (#757575)
 - Current months: `color.text.strong` (#333333)

**Comment by halocline** [comment]
@luketa8 @jcfilben Thank you for including the success criteria and the evaluations for each option.

Recommendations look good to me. @taysea can you do a final review and perhaps share with @gregfuruiye?

**Comment by taysea** [comment]
This looks good, Greg and Daniel approve of the proposal.

**Comment by jcfilben** [comment]
Closing this ticket. Follow on work tracked here: https://github.com/grommet/grommet/issues/7467

**Comment by luketa8** [comment]
Updated [Success Criteria](https://www.figma.com/design/OBxHRrYylrg1IyNetpK7OF/%237423---Calendar-contrast?node-id=80-886&t=8xu3uvtu8rnNyEiZ-11)

Added proposal options for [Light mode](https://www.figma.com/design/OBxHRrYylrg1IyNetpK7OF/%237423---Calendar-contrast?node-id=145-153&t=B11GZ8fgPYkzndpB-11) and [Dark mode ](https://www.figma.com/design/OBxHRrYylrg1IyNetpK7OF/%237423---Calendar-contrast?node-id=147-401&t=B11GZ8fgPYkzndpB-11). 


## Grommet theme

### Light mode:

 - Leading/trailing month days: `text-xweak` (#666666)
 - Current month days: `text-strong` (#000000)
 - Weekdays: `text` (#444444)
 
### Dark mode

 - Leading/trailing month days: `text-xweak` (#BBBBBB)
 - Current month days: `text-strong` (#FFFFFF)
 - Weekdays: `text-weak` (#CCCCCC)

**Comment by jcfilben** [comment]
Thanks for the updated color proposal! Do you have something for the HPE theme as well or is that part still in progress?

**Comment by luketa8** [comment]
Currently WIP, I'll update shortly.

**Comment by luketa8** [comment]
Added proposal options for HPE Theme, [Light mode](https://www.figma.com/design/OBxHRrYylrg1IyNetpK7OF/%237423---Calendar-contrast?node-id=157-1533&t=cFGdz1ju5koCG0j0-11) and [Dark mode ](https://www.figma.com/design/OBxHRrYylrg1IyNetpK7OF/%237423---Calendar-contrast?node-id=157-1457&t=cFGdz1ju5koCG0j0-11). 
 
 In this case, we lack visual distinction between `color.text.default` and color.text.strong` tokens in Dark mode. 

## HPE theme

### Light mode:

 - Weekdays: `color.text.weak` (#757575)
 - Leading/trailing month days: `color.text.default` (#555555)
 - Current month days: `color.text.strong` (#333333)

 
### Dark mode

 - Weekdays: `color.text.weak` (#FFFFFF80)
 - Leading/trailing month days: `color.text.default` (#FFFFFF)
 - Current month days: `color.text.strong` (#FFFFFF)

**Comment by jcfilben** [comment]
Thanks for putting these together! I feel like the days of the week color is a subtle difference. I understand we only have so many color options to work with so it's tricky.

@halocline what level of prominence were you hoping for? Are the above designs meeting what you wanted or we you expecting the days of the week (M, T, W...) to be more prominent than the calendar dates that are in the month, or something else?

**Comment by halocline** [comment]
> [@halocline](https://github.com/halocline) what level of prominence were you hoping for? Are the above designs meeting what you wanted or we you expecting the days of the week (M, T, W...) to be more prominent than the calendar dates that are in the month, or something else?

Current month days = text.strong, correct?

I'm not sure that having three text colors in so close proximity helps.

Can we see what these options look like?

- Option 2 --> days of week = color.text.strong
- Option 3 --> days of week = color.text.weak

**Comment by jcfilben** [comment]
> Current month days = text.strong, correct?

In the current calendar the current month days are using `text` (#555555) not `text-strong`

**Comment by luketa8** [comment]
Added some [Research](https://www.figma.com/design/OBxHRrYylrg1IyNetpK7OF/%237423---Calendar-contrast?node-id=187-2164&t=dzdvi6SiVbJUka7M-11) around four different design system's datepickers. 

Most of them use the same text color for weekdays and current month's days and a lighter color for leading/trailing month's days. Even the DS that use three colors don't provide a real perceived distinction between the three, which is really hard to achieve in the human eye with text colors (except maybe with green). 

I'll update my proposal shortly based on these conclusions.

**Comment by luketa8** [comment]
Added new options for

- [Grommet light mode](https://www.figma.com/design/OBxHRrYylrg1IyNetpK7OF/%237423---Calendar-contrast?node-id=145-153&t=mfWVs4v9ZFaR4P9v-11)
- [Grommet dark mode](https://www.figma.com/design/OBxHRrYylrg1IyNetpK7OF/%237423---Calendar-contrast?node-id=147-401&t=mfWVs4v9ZFaR4P9v-11)

- [HPE light mode](https://www.figma.com/design/OBxHRrYylrg1IyNetpK7OF/%237423---Calendar-contrast?node-id=157-1533&t=mfWVs4v9ZFaR4P9v-11)
- [HPE dark mode](https://www.figma.com/design/OBxHRrYylrg1IyNetpK7OF/%237423---Calendar-contrast?node-id=157-1457&t=mfWVs4v9ZFaR4P9v-11)

These are the options that I'd propose:

## Grommet theme

### Light mode:

 - Weekdays: `text-strong` (#000000)
 - Leading/trailing month days: `text-xweak` (#666666)
 - Current month days: `text-strong` (#000000)

### Dark mode:

 - Weekdays: `text-strong` (#FFFFFF)
 - Leading/trailing month days: `text-xweak` (#BBBBBB)
 - Current month days: `text-strong` (#FFFFFF)

--

## HPE theme

### Light mode:

 - Weekdays: `color.text.strong` (#333333)
 - Leading/trailing month days: `color.text.weak` (#757575)
 - Current month days: `color.text.strong` (#333333)

 
### Dark mode

 - Weekdays: `color.text.strong` (#FFFFFF)
 - Leading/trailing month days: `color.text.weak` (#FFFFFF80)
 - Current month days: `color.text.strong` (#FFFFFF)

**Comment by jcfilben** [comment]
I agree with the options you indicated as being your favorite. (Looks like this is option 3 in most cases). Also agree with leaning away from trying to use 3 different gray's and rather just stick to 2 colors for this.
 
Tagging @halocline, let us know your thoughts.

**Comment by halocline** [comment]
> Tagging [@halocline](https://github.com/halocline), let us know your thoughts.

I am good with the recommendations and like that we are using two text values instead of three.

**Comment by jcfilben** [comment]
Closing this issue, I think we can move forward with the proposed colors

**Comment by jcfilben** [comment]
Follow on work tracked here: https://github.com/grommet/grommet/issues/7467
